src/main/java/sheets/MetadataSheetParser.java

```diff
@@ -0,0 +1,224 @@
+package sheets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.google.api.services.sheets.v4.Sheets;
+import com.google.api.services.sheets.v4.model.ValueRange;
+
+import dtd.Field;
+import dtd.Section;
+import dtd.TypeSpecificElement;
+
+public class MetadataSheetParser {
+
+    private Sheets service;
+    private String spreadsheetId;
+    private static final String GENERAL_INFORMATION_SHEET = "General Information";
+    private static final String TRANSCRIPTOMICS_SHEET = "Transcriptomics";
+    private static final String PROTEOMICS_SHEET = "Proteomics";
+    private static final String VALUES_SHEET = "Drop-down Values";
+    private static final String DATA_TYPE_SHEET = "Data Types";
+
+    private static final String METADATA_MODULE_COLUMN_KEY = "Metadata Module";
+    private static final String LABEL_COLUMN_KEY = "Property";
+    private static final String TYPE_COLUMN_KEY = "Property Type";
+    private static final String FIELD_NAME_COLUMN_KEY = "Field Name";
+    private static final String REQUIRED_COLUMN_KEY = "Required?";
+    private static final String VALIDATIONS_COLUMN_KEY = "Validations?";
+    private static final String LINKED_WITH_COLUMN_KEY = "Linked With";
+    private static final String DISPLAY_WHEN_COLUMN_KEY = "Display When";
+    private static final String CONSTRAINTS_COLUMN_KEY = "Constraints";
+    private static final String ADDITIONAL_PROPS_COLUMN_KEY = "Additional Props";
+
+    private static final String CELL_RANGE_NOHEADER = "1:999";
+    private static final String CELL_RANGE_HEADER = "2:999";
+
+
+    public MetadataSheetParser(Sheets service, String spreadSheetId) throws IOException {
+        this.service = service;
+        this.spreadsheetId = spreadSheetId;
+    }
+
+    public Map<String, TypeSpecificElement> getTypeSpecificElements() throws IOException {
+        Map typeSpecificElements = new HashMap<String, TypeSpecificElement>();
+        String range = getRange(DATA_TYPE_SHEET, CELL_RANGE_HEADER);
+        List <List<Object>> rows = getRows(range);
+        for (List row: rows) {
+            TypeSpecificElement element = new TypeSpecificElement();

```

Comment:
This is not a problem now, but we can make this more efficient if we create a custom constructor that takes the values in rather than using setters.  I am not sure what the performance is on this right now, but just something to keep in mind if we start to see poor performance.